fuel-storage/src/error.rs

```diff
@@ -0,0 +1,55 @@
+/// Wraps around possible errors that can occur during storage operations.
+#[derive(Debug)]
+#[cfg_attr(feature = "std", derive(thiserror::Error))]
+pub enum StorageError {
```

Comment:
Why do we want to move this error from `fuel-core` to `fuel-vm`?

Reply:
The main issue here is that if the error type lives in `fuel-core`, the `fuel-vm` cannot understand this type without depending on `fuel-core-storage` which would be circular. This error is used in `InterpreterError`. We could instead use `anyhow::Error` or a custom error trait, but those have their own issues.

Reply:
I've changed the approach, see https://github.com/FuelLabs/fuel-vm/pull/582#issuecomment-1728570111